PCF-ADR-0003-codeflare-release-process.md

```diff
@@ -0,0 +1,123 @@
+    1. Make sure that any CRD updates in the MCAD repository are copied into the CodeFlare operator repository
+    2. Once merged, create a new operator image.
+3. Generate new operator bundle in CodeFlare Operator
+    1. Update the image reference for the bundle
+    2. As part of this, we will be testing the latest component versions
+    3. Update support matrix in operator README.
+4. Create tag in operator repository and update release notes of operator version to include new support matrix.
+5. Open a pull request to OpenShift community operators repository with latest bundle.
+6. Once merged, update component stable tags to point at the latest image release.
+7. Announce the new release in slack and mail lists, if any.
+8. Update the Distributed Workloads component in ODH.
+    1. At a minimum, we will need to update the README for Distributed Workloads to refer to the latest versions of each component
+    2. Updates could include custom resources, KubeRay version, or usage/installation instructions.
+    3. Update needs to occur within a week of CodeFlare release
+    4. Make sure that the platform SIG is aware of these updates, so that the ODH release notes are updated as expected
+
+Hotfix Process?
+If a critical issues is discovered requiring an out-of-band release, we will follow the normal release process, bumping up the z stream version for Project CodeFlare as well as the related component.
+
+## Open Questions
+
+1. How does versioning and releases for the distributed workloads component for ODH fit into this?
+    * Currently thinking it will just follow Project CodeFlare versions.
+2. Do we want to include a code/feature freeze date?
+    * Leaning towards keeping this window as small as possible – something like 2 days before the end of a given sprint.
+3. There’s work ongoing to include ui integrations for MCAD in the odh-dashboard. How does that fit into our release process?
+4. There’s a lot of scope for automation improvements as well as testing additions. How do we prioritize these inclusions?
+5. The CodeFlare operator currently uses pinned versions for component references, should we just use the component "stable" tags instead?
+    * Currently thinking "No" -- using explicit version references makes it very clear what version of what component is included in a given release of the stack.
+
+## Alternatives
+
+We didn't consider any other alternatives
+
+## Stakeholder Impacts
+
+| Group                         | Key Contacts       | Date       | Impacted? |
+| ----------------------------- | ------------------ | ---------- | --------- |
+| CodeFlare SDK                 | Mustafa Eyceoz     | 04/28/2023  | yes       |
+| MCAD                          | Abhishek Malvankar | 04/28/2023  | yes       |
+| InstaScale                    | Abhishek Malvankar | 04/28/2023  | yes       |
+| CodeFlare Operator            | Anish Asthana      | 04/28/2023  | yes       |
```

Comment:
Just from reading this over. It seems like this is very focused on the `CodeFlare Operator`.

It is mainly focused on:
A) How to update to specific versions of the codeflare operator's dependencies
B) How to make those changes available in ODH
C) How to disseminate news of the new operator version.

Does this just effect the other components because it creates a requirement for unit tests and e2e tests?

Reply:
The doc may feel that way since we are treating the CodeFlare operator as analogous to the entire Project CodeFlare stack.

```
* CodeFlare Operator
  * The CodeFlare Operator version will be aligned with (i.e equivalent to) the Project CodeFlare version
```

Reply:
> 

What does the Project CodeFlare version represent?

Reply:
The overall version of the stack. This will be pretty directly aligned with the Distributed Workloads version.